### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by davidabrahams**

### CODE REVIEW

**Comment by davidabrahams** [comment]
Things accomplished in this PR:

- `getTraversableTiles()` function was moved to a separate `helpers.js` file
- `self` variable refactored to `me`, since `self` is a reserved word
- wrote our first unit test, in `tests/test.js`. We can use this framework to built out unit tests for our code

File: README.md [hunk 1]
```diff
@@ -7,8 +7,11 @@ curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | b
 2. Restart your bash session and install node using NVM: `nvm install node`
 3. In this directory, install the node packages, including webpack: `npm install`
 
+## Testing the App
+- Run `npm test`. This will lint your test files, and then run them.
+
 ## Running the App
-- Run `npm run build`.  This will start webpack, which will "watch" the files in `src/` and will update the bundle automatically.  This will also run eslint whenever your code changes so you can see if the linter fails.
+- Run `npm run build`.  This will start webpack, which will "watch" the files in `src/` and will update the bundle automatically.  This will also run eslint whenever your code changes so you can see if the linter fails. It will also automatically run unittests whenever you save a file it is watching.
```
Comment (davidabrahams) [inline README.md]: dank

File: README.md [hunk 2]
```diff
@@ -7,8 +7,11 @@ curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | b
 2. Restart your bash session and install node using NVM: `nvm install node`
 3. In this directory, install the node packages, including webpack: `npm install`
 
+## Testing the App
+- Run `npm test`. This will lint your test files, and then run them.
+
 ## Running the App
-- Run `npm run build`.  This will start webpack, which will "watch" the files in `src/` and will update the bundle automatically.  This will also run eslint whenever your code changes so you can see if the linter fails.
+- Run `npm run build`.  This will start webpack, which will "watch" the files in `src/` and will update the bundle automatically.  This will also run eslint whenever your code changes so you can see if the linter fails. It will also automatically run unit tests whenever you save a file it is watching.
```
Comment (shanek21) [inline README.md]: I'm not sure how I feel about it running unit tests after every file change.

Especially when it does this to my terminal 
screenshot from 2017-07-12 22 57 14

Comment (davidabrahams) [inline README.md]: resolved via 5e3bbaa

Comment (shanek21) [inline README.md]: screenshot 2017-07-13 06 40 34

Should I do something besides just running `npm run build`?

Comment (davidabrahams) [inline README.md]: nope I made a mistake and left an import behind you don't need. Thanks for testing this. will fix

Comment (davidabrahams) [inline README.md]: @shanek21 pull and try again. 435542ef886 should have fixed this

Comment (shanek21) [inline README.md]: That fixed it. Good work!

**Pull request merged**